main/zoe/api/Zoe-Intro.md

```diff
@@ -0,0 +1,109 @@
+# Introduction to Zoe
+
+
+**Note:** Before reading this, you should be familiar with ERTP basics. (**tyg todo**: Add link)
+
+
+This is an introduction to Zoe, Agoric's smart contract framework. The Zoe Guide (**tyg todo**: Add link) has more complete and detailed information, including concepts not covered here. In particular, here we simplify Zoe examples such that there are always two parties in a basic trading/swap contract. We do not discuss more complicated contracts with many participants, such as auctions, or simpler contracts, such as our AutomaticRefund one party demo contract.
+
+
+## What is Zoe?
+
+
+*Zoe* is itself a long lived and trusted contract that serves as a framework which runs other contracts.
+
+
+## How do Zoe contracts work?
+
+
+To start using a contract, you need to *install* it on Zoe. This is putting the contract’s code into Zoe. Multiple contracts can be installed on Zoe, and each contract can have multiple simultaneous active offers, each of which is handled by a separate contract instance. For clarity, in this document a *contract* is the installed code that defines how the contract works. A *contract instance* is an active execution of the contract code.
+
+
+Once installed on Zoe, a contract waits for an *offer* to be made. That starts a lifecycle for the offer and new contract instance of:
+1. Sending out *invites* to request counter-offers.
+2. Partie *redeeming* those invites and making a counter-offer.
+3. Determining when the offer *matches* a counter-offer.
+4. Either succeeding or failing to make the transaction. 
+
+
+An offer’s completion only affects that offer and the assets involved in it; essentially its contract instance. The contract it was made on keeps running, waiting for and processing new offers. 
```

Comment:
There is *not* a contract instance per offer. There is a contract instance per `makeInstance`. Within a contract instance, multiple offers are matched. Only contract instances keep running. A contract (installation) is just code.

Reply:
OK, this is a tricky bit. This was my attempt to address issues that Kate brought up about an earlier draft about how to phrase/describe when things stop/end. So we've got an installed contract, which is the code. That never ends, at least as long as Zoe and/or the server is running. Then we have a contract instance. If I understood Kate correctly, it's not right say that ends, as it's really the first offer within the instance that gets the process going.  So is it "first offer" finishes?  

What I'm trying to ask here is that we have a lifecycle kicked off by a combo of instantiating a contract and creating a contract instance. But that's still just sitting there, not doing anything, until someone takes the invite generated by the instantiation and uses it to make an initial offer, then makes other invites and sends them out, etc.  At some point, things stop because the contract code is satisfied that the transaction is finished and (assume a success) everyone has what they should have at the end.    But I don't know what the appropriate way is to describe that things are done for this execution (in a legal sense) of the contract?

Reply:
@katelynsills by `makeContract` above, did you mean `makeInstance`.

Also, rather than `zoe.make...` it should be `E(zoe).make...`.

Reply:
Yup, I did mean `zoe.makeInstance`.